src/hotspot/cpu/x86/x86.ad

```diff
@@ -9065,34 +9036,47 @@ instruct evcmp_masked(kReg dst, vec src1, vec src2, immI8 cond, kReg mask, rRegP
   ins_pipe( pipe_slow );
 %}
 
-instruct mask_all_evexI(kReg dst, rRegI src, vec xtmp) %{
+#ifdef _LP64
+instruct mask_all_evexI_imm(kReg dst, immI cnt, rRegL tmp) %{
+  match(Set dst (MaskAll cnt));
+  effect(TEMP_DEF dst, TEMP tmp);
+  format %{ "mask_all_evexI $dst, $cnt \t! mask all operation" %}
+  ins_encode %{
+    int vec_len = vector_length(this);
+    __ movq($tmp$$Register, $cnt$$constant);
+    __ kmovql($dst$$KRegister, $tmp$$Register);
+    __ kshiftrql($dst$$KRegister, $dst$$KRegister, 64 - vec_len);
+  %}
+  ins_pipe( pipe_slow );
+%}
+
+instruct mask_all_evexI(kReg dst, rRegI src, rRegL tmp) %{
   match(Set dst (MaskAll src));
-  effect(TEMP xtmp);
+  effect(TEMP_DEF dst, TEMP tmp);
   format %{ "mask_all_evexI $dst, $src \t! mask all operation" %}
   ins_encode %{
-    int vlen_enc = vector_length_encoding(vector_length(this));
-    __ evpbroadcastb($xtmp$$XMMRegister, $src$$Register, vlen_enc);
-    __ evpmovb2m($dst$$KRegister, $xtmp$$XMMRegister, vlen_enc);
+    int vec_len = vector_length(this);
+    __ movslq($tmp$$Register, $src$$Register);
+    __ kmovql($dst$$KRegister, $tmp$$Register);
+    __ kshiftrql($dst$$KRegister, $dst$$KRegister, 64 - vec_len);
```

Comment:
Could we not do kmovdl followed by kshiftrdl here? Thereby removing the need for movslq.

Reply:
maskAll accept a boolean argument (false (0) , true(-1)).
value operand has a rRegI register class which represent 32 bit register. This value need to be sign extended to 64 bit value before computing the final mask value using shift right operation.